src/app/modules/users/add-multiple-users/add-multiple-users.component.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, OnInit, Optional, Inject } from '@angular/core';
+import { UsersService, CommonServiceService } from '../../admin-core';
+// import { saveAs } from 'file-saver';
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material';
+import { saveAs } from "file-saver";
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ConfirmDialogComponent, ConfirmDialogModel } from '../../admin-shared';
+
+
+
+@Component({
+  selector: 'app-add-multiple-users',
+  templateUrl: './add-multiple-users.component.html',
+  styleUrls: ['./add-multiple-users.component.scss']
+})
+export class AddMultipleUsersComponent implements OnInit {
+  filecontent: any;
+  downloadurl: 'download url from server';
+  myForm: FormGroup;
+  orgData: any;
+  requestId: any;
+  selected: any;
+  // submitClick: boolean = false;
+  confirmPopupResult: any;
+  onload = {
+    buttonName: 'UPLOAD',
+    submitClick: false
+  }
+  constructor(private usersService: UsersService,
+    private _snackBar: MatSnackBar,
+    public fb: FormBuilder,
+    private commonServiceService: CommonServiceService,
+    @Optional() @Inject(MAT_DIALOG_DATA) public data,
+    private dialog: MatDialog,
+    public dialogRef: MatDialogRef<AddMultipleUsersComponent>
+  ) { }
+
+  ngOnInit() {
+    console.log('mmmmmmmmmmmm', this.data.defaultValue);
+    this.selected = this.data.defaultValue;
+
+    // this.myForm.patchValue({ organisation: this.data.defaultValue });
+    this.reactiveForm();
+    this.getOrganisations();
+  }
+
+  reactiveForm() {
+    this.myForm = this.fb.group({
+      organisation: [this.data.defaultValue.value, [Validators.required]],
+      userscsv: ['', [Validators.required]]
+    })
+    this.myForm.controls['organisation'].setValue(this.data.defaultValue.value, { onlySelf: true });
+  }
+
+  /* Handle form errors*/
+  public errorHandling = (control: string, error: string) => {
+    return this.myForm.controls[control].hasError(error);
+  }
+
+  getOrganisations() {
+    this.usersService.getOrganisations().subscribe(data => {
```

Comment:
Why is this API call again here, Please pass the list from userlisting component

Reply:
Changed